sw-emulator/lib/cpu/src/csr_file.rs

```diff
@@ -102,7 +102,7 @@ impl Csr {
 /// Configuration and status register file
 pub struct CsrFile {
     /// CSRS
-    csrs: [Csr; CsrFile::CSR_COUNT],
+    csrs: Box<[Csr; CsrFile::CSR_COUNT]>,
```

Comment:
I'm okay with this, but if the CPU is boxed then I'm not sure this does much?

And I know it is the emulator, but a 4096-len array seems silly :) If we're pulling in standard library anyway, this feels like it should be a `HashMap` or a `Box<HashMap>`, or some other sparse data structure.

Reply:
This is all a bit of dance with the optimizer. Theoretically LLVM *should* be able to RVO all these non-result constructors, but it's not guaranteed so I decided to box some of the larger types to increase the chance that it does the right thing *somewhere*.